### DISCUSSION TITLE
How to make a `BaseModel` accept arguments by position like a `dataclass`?

### DISCUSSION THREAD

**Discussion by gbaian10** [pydantic/pydantic#10460] [Question]
```py
from dataclasses import dataclass

from pydantic import BaseModel


class MyBaseModel(BaseModel):
    name: str
    age: int


@dataclass
class MyDataclass:
    name: str
    age: int


a = MyBaseModel(name="John", age=30)  # ok
b = MyDataclass("John", 30)  # ok
c = MyBaseModel("John", age=30)  # TypeError
```

How can I make a `BaseModel` accept arguments by position like a `dataclass`?
If possible, is there a way to allow only some arguments, such as the first one, to be positional, while the others are not (similar to using "/" in a function)?

**Comment by Viicos** [comment]
> How can I make a `BaseModel` accept arguments by position like a `dataclass`?

We currently do not support positional arguments for Pydantic models, although technically it could be possible.

> If possible, is there a way to allow only some arguments, such as the first one, to be positional, while the others are not (similar to using "/" in a function)?

Python dataclasses have the [`KW_ONLY`](https://docs.python.org/3/library/dataclasses.html#dataclasses.KW_ONLY) sentinel value to do so. Similarly, it could be technically possible to support this in Pydantic.

This is considered low prio for now.

**Reply by gbaian10** [reply]
@Viicos 
I later found a simple way to implement this function, but I'm not sure if there will be any side effects, or if you would advise against using this approach.

```py
from typing import Any

from pydantic import BaseModel


class MyBaseModel(BaseModel):
    name: str
    age: int

    def __init__(self, name: str, age: int, **kwargs: Any) -> None:
        super().__init__(name=name, age=age, **kwargs)


a = MyBaseModel(name="John", age=30)
b = MyBaseModel("John", age=30)
c = MyBaseModel("John", 30)
```
